plugins/kubernetes/src/utils/owner.ts

```diff
@@ -54,17 +54,24 @@ export const getOwnedPodsThroughReplicaSets = (
   }, [] as V1Pod[]);
 };
 
+interface ResourceRef {
+  kind: string;
+  namespace?: string;
+  name?: string;
+}
+
 export const getMatchingHpa = (
-  ownerName: string | undefined,
-  ownerKind: string,
+  owner: ResourceRef,
   hpas: V1HorizontalPodAutoscaler[],
 ): V1HorizontalPodAutoscaler | undefined => {
   return hpas.find(hpa => {
     return (
       (hpa.spec?.scaleTargetRef?.kind ?? '').toLocaleLowerCase('en-US') ===
-        ownerKind.toLocaleLowerCase('en-US') &&
+        owner.kind.toLocaleLowerCase('en-US') &&
+      (hpa.metadata?.namespace ?? '') ===
+        (owner.namespace ?? 'unknown-namespace') &&
```

Comment:
So this is essentially `a.metadata?.namespace && owner.namespace && a.metadata.namespace === owner.namespace` now right? Or is it possible for `owner.namespace` to be `''`?

Does that logic make sense or should it more along the lines of `(!a.metadata?.namespace && !owner.namespace) || a.metadata.namespace === owner.namespace`?

Reply:
Namespace will never be null/undefined as HPA must be in a namespace and it can only target objects in its namespace.

I copied the same logic that we have for `name` there. The reason for having the null checks is because the model from Kubernetes has `name?: string` and `namespace?: string` because in other contexts, these could be null.